### ISSUE TITLE
Raft metrics disappear after upgrading to 1.6.4 

### ISSUE THREAD

**Issue by qk4l** [hashicorp/nomad#19369]
### Nomad version
Nomad v1.6.4
BuildDate 2023-12-07T08:27:54Z
Revision dbd5f36a24a924e2ba4dd6195af6a45c922ac8c6

### Operating system and Environment details
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04 LTS
Release:	22.04
Codename:	jammy

### Issue
After updating from 1.6.2 to 1.6.4 almost all raft metrics disappear from endpoint  `/v1/metrics`
Only `nomad_raft_appliedIndex` and `nomad_raft_lastIndex` exists.

### Reproduction steps
```
curl https://$NOMAD_URL/v1/metrics?format=prometheus | grep raft
# HELP nomad_raft_appliedIndex nomad_raft_appliedIndex
# TYPE nomad_raft_appliedIndex gauge
nomad_raft_appliedIndex{host="NOMAD_URL"} 3.748674e+06
# HELP nomad_raft_lastIndex nomad_raft_lastIndex
# TYPE nomad_raft_lastIndex gauge
nomad_raft_lastIndex{host="NOMAD_URL"} 3.748674e+06
```
#### Expected Result
More metrics at least
`nomad.raft.leader.lastContact`, `nomad.raft.replication.appendEntries` 
which is described in  [docs](https://developer.hashicorp.com/nomad/docs/v1.6.x/operations/monitoring-nomad#consensus-protocol-raft).

#### Actual Result
Only two metrics exists
```
# HELP nomad_raft_appliedIndex nomad_raft_appliedIndex
# TYPE nomad_raft_appliedIndex gauge
nomad_raft_appliedIndex{host="NOMAD_URL"} 3.748674e+06
# HELP nomad_raft_lastIndex nomad_raft_lastIndex
# TYPE nomad_raft_lastIndex gauge
nomad_raft_lastIndex{host="NOMAD_URL"} 3.748674e+06
```

**Comment by tgross** [comment]
Hi @qk4l! I was able to confirm this happened between 1.6.3 and 1.6.4. The most suspicious commit is https://github.com/hashicorp/nomad/pull/19190 where we changed out the library version for go-metrics. I'll investigate further and report back.

**Comment by atillamas** [comment]
Not only raft metrics. If you do a `curl <nomad-server>:4646/v1/metrics?format=prometheus | wc -l` it will go from 536 to 225 or something (cannot remember since i reverted version back to 1.6.3)

**Comment by tgross** [comment]
Yeah the 0.5.0 version of go-metrics changes the import path, which needs to be done in all dependencies as well, not just the binary (due to a very unfortunate use of a package-global for state). We've reverted that change and it'll go out in the next patch. Meanwhile, we'll figure out how to fix this more generally.

**Comment by tgross** [comment]
Apparently we messed up the backport process and this landed only in the 1.6.x and 1.5.x backports (which haven't shipped yet) and not the 1.7.1 release we just did. :facepalm: It will go out in the _next_ patch, which will likely be fairly soon once we've got a few other bugs figured out.

**Comment by tgross** [comment]
Our revert patch for this issue has landed and will be shipped in Nomad 1.7.2 shortly (likely tomorrow).
